### ISSUE TITLE
Button + Selector

### ISSUE THREAD

**Issue by alejandroivan** [yannickl/Reactions#16]
How do I know when the reaction changed on a button? Even if that's triggered from a selector?

**Comment by alejandroivan** [comment]
Nevermind I managed to make it work. The key is to catch `UIControl` actions and mix the result with the `isSelected` property (saving the last reaction selected in the selector). So basically:

```
// UIView with class ReactionButton in storyboard, connected to this outlet
@IBOutlet var reactionButton: ReactionButton!

// We have to know what the last focused reaction was, since selectorReaction is null at the beginning
private var lastFocusedReaction: Reaction?


func setupReactionsSelector() {
    let reactionSelector = ReactionSelector()
    reactionSelector.reactions = Reaction.facebook.all

    // This one takes a value from ReactionSelector
    reactionButton.addTarget(self, action: #selector(reactionChanged), for: .valueChanged)
    // And this one from ReactionButton
    reactionButton.addTarget(self, action: #selector(reactionChanged), for: .touchUpInside)

    // The last focused reaction is empty, so we'll make it the first of the reactions
    lastFocusedReaction = reactionSelector.reactions.first
}

func reactionChanged(_ sender: AnyObject) {
    guard let reaction = self.reactionButton.reactionSelector?.selectedReaction else {
        // No reaction here. This happens when tapping the button.
        // We'll use the selected state to know if the content was liked or unliked (or other reaction).
        if let reaction = lastFocusedReaction {
            if ( reactionButton.isSelected ) {
                // The current reaction was selected (liked, loved, wow-ed, etc.)
                // Here you'll want to inform the delegate that a reaction was chosen
            }
            else {
                // The content was unliked, un-haha-ed, un-wow-ed, whatever.
                // Check lastFocusedReaction for the correct reaction that was deselected.
                // Here you'll want to inform the delegate that a reaction was chosen
            }
        }
        
        // Nothing to do here.
        return;
    }

    lastFocusedReaction = reaction

    // Reaction was selected through ReactionSelector()
    // Here you'll want to inform the delegate that a reaction was chosen
}
```

**Comment by yannickl** [comment]
Hi @alejandroivan, I don't really understand your issue.

Do you want to react to a reaction selection?

**Comment by alejandroivan** [comment]
@yannickl don't worry, I already solved it.

What I wanted to accomplish was to mimick Facebook reactions button behavior, using a ReactionButton with a ReactionSelector. I was able to detect when ReactionSelector changed the selected reaction (while sliding my finger), but I needed to detect it like a "touch up" event (when lifting my finger).

Also detect when touching the ReactionButton (touch up), when it was selected (reaction added) or deselected (reaction removed).

I made it possible with the code I posted. Maybe it's useful for someone else.
